### ISSUE TITLE
508-defect-2 [SCREENREADER]: Both forms' radio groups must be refactored to use standard markup

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#26579]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Team PM will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a screen reader user, I want to hear the radio buttons announced consistently. This includes the legend text on the first radio button when it receives keyboard or virtual cursor focus, and the individual labels as I move through the radio group.

## Details

Radio groups are using a `<label>` for the group text instead of a `<legend>`, custom markup for individual labels. These need to be refactored to a proper fieldset, legend, radio button pattern. Screenshot and code snippet below.

## Acceptance Criteria

- [ ] Checkbox HTML is properly refactored
- [ ] Radio groups announce as expected to screen readers
- [ ] 0 axe-core violations

## Proposed Solution (if known)

Let's remove the extra ARIA markup and use a standard markup pattern. I've excluded the Angular-specific markup for brevity.

```html
<fieldset>
  <legend>Were you able to do your task today?</legend>
  <div>
    <label for="task-answer-1">
      <input type="radio" id="task-answer-1" name="tasks" value="yes">
      Yes
    </label>
    <label for="task-answer-2">
      <input type="radio" id="task-answer-2" name="no">
      No
    </label>
  </div>
  ...
</fieldset>
```

## WCAG or Vendor Guidance (optional)
* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

## Screenshots or Trace Logs

![Screen Shot 2021-06-24 at 12 05 40 PM](https://user-images.githubusercontent.com/934879/123305343-9eb20c00-d4e5-11eb-8c00-0c65c0f017b3.png)

**Comment by pjhill** [comment]
Sarkis indicated this was previously discussed, and this was not deemed a 508 issue upon previous examination. Do we need to discuss this in more detail with Medallia configuration team?

**Comment by 1Copenut** [comment]
It was passed as a non-issue the first time in a quick spot test. Further testing under more realistic conditions gave me pause, and why I entered this new ticket. The labels for individual radio buttons are verbose -- each one reads the legend text, and the individual label. A fieldset + legend + input/label pattern would read the legend once, before the first input, then just read the input labels. This is enough of a cognitive overhead I felt it worth revisiting.
